docs/SCHEMA.md

```diff
@@ -0,0 +1,348 @@
+# .env File Schema
+
+`configure-env` supports any valid `.env` file that can be parsed by libraries like [`dotenv`](https://npm.im/dotenv).
+
+That means it supports:
+
+- "variable declaration": key value pairs in the pattern: `KEY=VALUE`
+- empty declarations are valid but **must** contain a `=`. Example: `KEY=`
+- one key value pair per line
+- empty lines are ignored
+- lines beginning with # are treated as comments
+
+Based on those rules, [special comment annotations](#special-comment-annotations) and [best effort guessing](#best-effort-guessing), this tool will extract meta information for each key value pair that can either be extracted using the `parse` or `parseFile` function or that will be used by the CLI component of `configure-env` to prompt users for input of said environment variables.
+
+## Special Comment Annotations
+
+Any comments that are immediately before a "variable declaration" will be evaluated in the context of the variable declaration. Special comments can be used to define:
+
+- [`required`](#required)
+- [`format`](#format)
+- [`description`](#description)
+- [`link`](#link)
+- [`default`](#default)
+- [`configurable`](#configurable)
+
+These comments can be used in any order as long as they are right before the variable declaration with no empty line before it.
+
+### `required`
+
+Using `# required: true` or `# required: false` can define whether a variable has to be configured in order for the app to work.
+
+**Default Value**: `true`
+
+**Example `.env` file:**
+
+```bash
+# required: true
+TWILIO_ACCOUNT_SID=
+```
+
+**Parsed Result (with omitted default values):**
+
+```json
+[{ "key": "TWILIO_ACCOUNT_SID", "required": true }]
+```
+
+### `format`
+
+A format comment can be used to inform a user what types of value is being expected for a variable.
```

Comment:
So you want a format and a type? The problem is that essentially at the end every environment variable in Node.js is a `string`. So this would largely be format validation during parsing.

Reply:
fair enough